.githooks/pre-commit/update-templates

```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+
+if ! echo "$STAGED_FILES" | grep -q "base-template-symlink|update-templates"; then
+    # stop here if there are no changes for this commit
+    exit 0
+fi
+
+# Patch the hooks
+HOOK_NAMES=$(sed -e '1,/MANAGED_HOOK_NAMES="/d;/"/,$d' install.sh)
+
+for HOOK_NAME in $HOOK_NAMES; do
+    if ! echo "$HOOK_NAMES" | grep -q "$HOOK_NAME"; then
+        echo "! You changed the managed hooks types" >&2
+        echo "  you need to adjust them in \`$0\`" >&2
+        exit 1
+    fi
+
+    cp base-template-symlink.sh "hooks/$HOOK_NAME"
+    if ! sed -i "s|INSTALL_DIR=.*|# Auto-generated: do not edit!\nINSTALL_DIR=\$(cd \"\$(dirname \"\$0\")/../\" \&\& pwd)|" "hooks/$HOOK_NAME" ||
```

Comment:
Ehm, did you note that the hooks in `hooks/` are only here for a quick install (as @matthijskooijman suggested) I though of having these checked in for that sole purpose...

We actuall still install the wrappers `base-template-symlink.sh` into the repos `.git/hooks` dir.

Do you mean when we place the wrapper `base-template-symlink.sh` in the repos, we directy replace the INSTALL_DIR? Does that make sense, I rather have looked it up in the config... ?

Reply:
Will that even work? Won't we need the install dir settings to exist when running hooks? Like for checking for updates, etc. Maybe we should just do the symlink change here, but ignore the pre-generated hooks in this first round, so we can see how those work properly in a next PR? What do you think?

Reply:
I'm still hoping githooks can be made to run without any installation other than setting `core.hooksPath`, but then indeed `INSTALL_DIR` must be derived from the script location rather than git config, so this replace seems reasonable to me (since the pregenerated version needs the derive-from-$0 while the installed version needs get-from-git-config).

But indeed, this is rather orthogonal to the change in this PR (unless this change would need changes in the symlink and more importantly the symlink-to-base-template calling interface, since that cannot be easily changed once merged, I think).